### PR TITLE
CI build changed plugins

### PR DESCRIPTION
**Pull request by longzheng**
Add GitHub Actions workflow for any PRs that changes /plugins to build any plugins that changed

Fix snapshot test script

### CODE REVIEW

File: plugins/snapshot/test/test.ts [hunk 1]
```diff
@@ -27,29 +26,4 @@ async function main1() {
     console.log('test1 done');
 }
 
-async function main2() {
-    const ret = await sharpFilterImage('/Users/koush/Downloads/151-1678381127261.jpg',
```
Comment (longzheng) [inline plugins/snapshot/test/test.ts:31]: Removed this since `sharpFilterImage` no longer exists

File: plugins/snapshot/test/test.ts
Comment (longzheng) [inline plugins/snapshot/test/test.ts:1]: I presume you used this file to play around with ffmpeg, so I left it

File: npm-install-base.sh
Comment (longzheng) [inline npm-install-base.sh:1]: I'm not very experienced with shell scripts so I'm not sure if this is the ideal way to split up this script

**Comment by koush** [comment]
is this ready?

**Comment by longzheng** [comment]
> is this ready?

Yes it is

**Pull request merged**